addons/account/models/account_journal.py

```diff
@@ -600,9 +600,19 @@ def _alias_prepare_alias_name(self, alias_name, name, code, jtype, company):
         )
         if company != self.env.ref('base.main_company'):
             company_identifier = company.name if self.env['mail.alias']._is_encodable(company.name) else company.id
-            if f'-{company_identifier}' not in alias_name:
-                alias_name = f"{alias_name}-{company_identifier}"
-        return self.env['mail.alias']._sanitize_alias_name(alias_name)
+            if f'-{company_identifier}' not in base_alias_name:
+                base_alias_name = f'{base_alias_name}-{company_identifier}'
+
+        # Check for duplicate alias, if exists append a suffix to ensure uniqueness
+        # It's not a duplicate if it's the same journal
+        sanitized_alias_name = self.env['mail.alias']._sanitize_alias_name(base_alias_name)
+        existing_alias = self.env['mail.alias'].search([('alias_name', '=', sanitized_alias_name)])
```

Comment:
That's not true anymore with introduction of alias domains, alias names are not unique anymore, but unique in a given alias domain.